spring-integration-core/src/main/java/org/springframework/integration/transformer/SimpleFromAvroTransformer.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.transformer;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificRecord;
+
+import org.springframework.beans.factory.BeanClassLoaderAware;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.integration.context.IntegrationContextUtils;
+import org.springframework.integration.expression.FunctionExpression;
+import org.springframework.integration.transformer.support.AvroHeaders;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+
+/**
+ * An Apache Avro transformer to create generated {@link SpecificRecord} objects
+ * from {@code byte[]}.
+ *
+ * @author Gary Russell
+ * @since 5.2
+ *
+ */
+public class SimpleFromAvroTransformer extends AbstractTransformer implements BeanClassLoaderAware {
+
+	private final Class<? extends SpecificRecord> defaultType;
+
+	private final DecoderFactory decoderFactory = new DecoderFactory();
+
+	private final Map<String, Class<? extends SpecificRecord>> typeCache = new ConcurrentHashMap<>();
```

Comment:
Shouldn't this be based on some eviction policy?
On the other hand isn't there a cache already on `ClassLoader` level, so any subsequent `ClassUtils.forName()` doesn't have impact?

Reply:
Good point